collections/collections.go

```diff
@@ -89,6 +89,25 @@ type KeyCodec[T any] interface {
 	Stringify(key T) string
 	// KeyType returns a string identifier for the type of the key.
 	KeyType() string
+
+	// MULTIPART keys
+
+	// EncodeNonTerminal writes the key bytes into the buffer.
+	// EncodeNonTerminal is used in multipart keys like Pair
+	// when the part of the key being encoded is not the last one,
+	// and there needs to be a way to distinguish after how many bytes
+	// the first part of the key is finished. The buffer is expected to be
+	// at least as big as SizeNonTerminal(key) returns. It returns
+	// the amount of bytes written.
+	EncodeNonTerminal(buffer []byte, key T) (int, error)
+	// DecodeNonTerminal reads the buffer provided and returns
+	// the key T. DecodeNonTerminal is used in multipart keys
+	// like Pair when the part of the key being decoded is not the
+	// last one. It returns the amount of bytes read.
+	DecodeNonTerminal(buffer []byte) (int, T, error)
+	// SizeNonTerminal returns the maximum size of the key K when used in
+	// multipart keys like Pair.
+	SizeNonTerminal(key T) int
```

Comment:
Instead of `*NonTerminal`, what about `*MultiPart`?

Reply:
The thing with multi-part is that both variants are used in multi-part keys, but some key parts are non-terminal and the last one isn't

Reply:
yeah, I share the same sentiment that `NonTerminal` suffixing expresses the concept better, I'm not sure if there's a more "common" name to express the concept.